src/feature/featuremodel.cpp

```diff
@@ -157,144 +157,116 @@ QObject* FeatureModel::get(const QString& feature) {
-          continue;
-        }
-
-        featuresFlippedOff.append(key);
-      }
+      featuresFlippedOff.append(key);
     }
-
-    settingsHolder->setFeaturesFlippedOn(featuresFlippedOn);
-    settingsHolder->setFeaturesFlippedOff(featuresFlippedOff);
   }
 
-#ifdef MZ_ADJUST
-  QJsonValue adjustFieldsValue = json["adjustFields"];
-  if (adjustFieldsValue.isUndefined()) {
-    logger.debug() << "No adjust fields found in feature list";
-    return;
-  }
+  settingsHolder->setFeaturesFlippedOn(featuresFlippedOn);
+  settingsHolder->setFeaturesFlippedOff(featuresFlippedOff);
+}
 
-  if (!adjustFieldsValue.isObject()) {
-    logger.error()
-        << "Error in the json format; adjust fields is not an object";
+// static
+void FeatureModel::updateExperimentalFeatures(
+    const QJsonValue& experimentalFeatures) {
+  if (!experimentalFeatures.isObject()) {
+    logger.error() << "Error in the json format: experimentalFeatures is"
+                      "not an object.";
     return;
   }
 
-  QJsonValue allowParameterValue = adjustFieldsValue["allowParameters"];
-  if (!allowParameterValue.isArray()) {
-    logger.error()
-        << "Error in the json format; allow parameters are not an array";
-    return;
-  }
+  QJsonObject experimentalFeaturesObj = experimentalFeatures.toObject();
+  QStringList experimentalFeaturesToToggleOn = experimentalFeaturesObj.keys();
 
-  QJsonArray allowParametersArray = allowParameterValue.toArray();
-  for (const QJsonValue& param : allowParametersArray) {
-    if (!param.isString()) {
-      logger.error()
-          << "Error in the json format; allowlist parameter is not a string";
+  for (const QString& key : experimentalFeaturesToToggleOn) {
+    const Feature* experimentalFeature = Feature::getOrNull(key);
+    if (!experimentalFeature) {
+      logger.warning() << "Got" << key
+                       << "but experimental feature doesn't exist. Ignoring.";
+      experimentalFeaturesToToggleOn.removeAll(key);
       continue;
     }
-    AdjustFiltering::instance()->allowField(param.toString());
-  }
 
-  QJsonValue denyParameterValue = adjustFieldsValue["denyParameters"];
-  if (!denyParameterValue.isObject()) {
-    logger.error()
-        << "Error in the json format; deny parameters in not an object";
-    return;
-  }
-
-  QJsonObject denyParameterObject = denyParameterValue.toObject();
-  for (const QString& key : denyParameterObject.keys()) {
-    QJsonValue value = denyParameterObject.value(key);
-    if (!value.isString()) {
-      logger.error()
-          << "Error in the json format; deny list parameter is not a string";
-      continue;
+    QJsonValue experimentalFeatureSettings = experimentalFeaturesObj[key];
+    if (!experimentalFeatureSettings.isObject()) {
+      logger.error() << "Error in the json format: experimentalFeature" << key
+                     << "is not an object.";
+      experimentalFeaturesToToggleOn.removeAll(key);
+      return;
```

Comment:
Should we continue instead of returning?

Reply:
Hm, idk. If one of them is wrong it's likely all are?

Reply:
We are continuing in some other error cases in this method. Up to you.